source/client/flush_worker_impl.h

```diff
@@ -0,0 +1,50 @@
+// Flush worker implementation. Flush worker periodically flushes metrics
+// snapshot to all configured stats sinks in Nighthawk.
+#pragma once
+
+#include <vector>
+
+#include "envoy/api/api.h"
+#include "envoy/event/dispatcher.h"
+#include "envoy/stats/sink.h"
+#include "envoy/stats/store.h"
+#include "envoy/thread_local/thread_local.h"
+
+#include "common/worker_impl.h"
+
+namespace Nighthawk {
+namespace Client {
+
+// Flush worker periodically flushes metrics snapshot to all configured stats
+// sinks in Nighthawk. A single flush worker is created in Nighthawk
+// process_impl.cc. It will keep running until exitDispatcher() gets called
+// after all client workers are completed in process_impl.cc. It will make the
+// last flush before shutdown in shutdownThread().
+class FlushWorkerImpl : public WorkerImpl {
+public:
+  FlushWorkerImpl(Envoy::Api::Api& api, Envoy::ThreadLocal::Instance& tls,
+                  Envoy::Stats::Store& store,
+                  std::list<std::unique_ptr<Envoy::Stats::Sink>>& stats_sinks,
+                  const std::chrono::milliseconds& stats_flush_interval);
+
+  void shutdownThread() override;
+
+  // Must be called in after all client workers are completed in
```

Comment:
Does this comment need updating to make it about FlushWorker? Also, tangentially related, do we want to ensure a single instance?

Reply:
Updated the comment.

By single instance, do you mean to make sure it is only called once?

Reply:
Yes, I think we don't ever want two instances to be alive at the same time, correct?

Reply:
Discussed offline. Instead of making it a singleton, for simplicity, decided to add comments that there should be only a single live flush worker instance at any given time in NH.